application/controllers/admin/Authentication.php

```diff
@@ -40,6 +40,17 @@ public function index()
             Yii::app()->session['adminlang'] = Yii::app()->request->getParam('loginlang', 'default');
             Yii::app()->setLanguage(Yii::app()->session["adminlang"]);
         }
+        /* Checking whether we have any templates to import */
+        if (($templatetoken = Yii::app()->request->getParam('templatetoken', 'default')) !== 'default') {
```

Comment:
Again, should already have been installed via CLI, or?

Reply:
Here we check for the existence of a token parameter in the URL and store it into the session, so later on the Survey site will know that a confirmation will needed to be displayed for the template.

![image](https://github.com/LimeSurvey/LimeSurvey/assets/2805488/e3c026eb-6bf6-4ab6-a9bf-2e94d31d7a6d)

Reference: https://limesurvey.atlassian.net/browse/MW-446